tests/test_core.py

```diff
@@ -5897,6 +5897,9 @@ def test_dlmalloc(self):
 
   @no_asan('asan also changes malloc, and that ends up linking in new twice')
   def test_dlmalloc_partial(self):
+    # Wasm backend warns when exception specifications with types are used, so
+    # suppress it
+    self.emcc_args += ['-Wno-wasm-exception-spec']
```

Comment:
Does this test rely on the `throw(std::bad_alloc)` thing? Can we just remove it?

Reply:
Yes I think we can and it can be simpler. Removed it.